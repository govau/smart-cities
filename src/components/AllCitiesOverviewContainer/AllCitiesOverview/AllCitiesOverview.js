/* eslint-disable */
import React from 'react';
import PropTypes from 'prop-types';
import PageWrapper from '../../PageWrapper/PageWrapper';
import CategoryOverviews from '../../CategoryOverviews/CategoryOverviews';
import Icon from '../../Icon/Icon';
import Link from '../../Link/Link';
import {
  ELEMENT_IDS,
  LINKS,
} from '../../../constants';
import smartCitiesLogo from './smart-cities-logo-white.svg';
import COLORS from '../../../style/_colors.scss';
import style from './AllCitiesOverview.scss';

const ncpLink = (
  <Link href={LINKS.CITIES_PERFORMANCE_NETWORK}>
    The National Cities Performance Framework
  </Link>
);

const smartCitiesLink = (
  <Link href={LINKS.SMART_CITIES_DPMC_LINK}>
    Smart Cities Plan
  </Link>
);

const AllCitiesOverview = props => (
  <PageWrapper>
    <div className={style.heroWrapper}>
      <div className={style.heroContainer}>
        <p className={style.heroText}>
          Tracking progress and performance across Australia’s largest cities
        </p>

        <img
          className={style.smartCitiesLogo}
          src={smartCitiesLogo}
          alt=""
        />
      </div>
    </div>

    <div className={style.subInfoWrapper}>
      <div className={style.subInfoContainer}>
        <h1
          id={ELEMENT_IDS.INITIAL_FOCUS_POINT_ON_PAGE}
          className={style.infoTitle}
        >
          Indicator Revision
        </h1>

        <p className={style.infoBodyText}>
          To keep the dashboard up-to-date, we've added more recent data on Australian cities. We've also changed the methods we use to calculate some of the indicators. So please be careful when interpreting any changes you notice.</p>
      </div>
    </div>

	 <div className={style.infoWrapper}>
      <div className={style.infoContainer}>
        <h1
          className={style.infoTitle}
        >
          About
        </h1>

        <p className={style.infoBodyText}>
          Australian cities have long been an important economic and social asset and will be
          more important than ever in the 21st century. Our cities provide great opportunities
          for innovation and productivity in the global knowledge-based economy and
          are recognised as great places to live, work and study.
        </p>

        <p className={style.infoBodyText}>
          The Australian Government is committed to the continual improvement of our
          cities. {ncpLink} will assist in this goal by providing data
          to help all levels of government, industry and the community to make the best
          policy and investment decisions for Australia’s future.
        </p>
      </div>
    </div>
 
    <div className={style.subInfoWrapper}>
      <div className={style.subInfoContainer}>
        <h1 className={style.infoTitle}>
          <span className={style.infoTitleHeavy}>Policy priorities</span> & Context
        </h1>

        <p className={style.subInfoBodyText}>
          There are 6 policy priorities to help achieve
          the objectives of the {smartCitiesLink}. The National Cities Performance
          Framework is designed to measure how well our cities are performing, using
          two types of indicators.
        </p>

        <dl className={style.legend}>
          <div className={style.legendRow}>
            <dt className={style.term}>
              <Icon
                className={style.icon}
                color={COLORS.GREY_600}
                icon="performanceIndicator"
                size={32}
              />
              Performance
            </dt>
            <dd className={style.definition}>
              Aspects that can be measured and tracked against policy priorities.
            </dd>
          </div>
          <div className={style.legendRow}>
            <dt className={style.term}>
              <Icon
                className={style.icon}
                color={COLORS.GREY_600}
                icon="contextualIndicator"
                size={32}
              />
              Context
            </dt>
            <dd className={style.definition}>
              Characteristics that help us understand what it’s
              like and why it functions the way it does.
            </dd>
          </div>
        </dl>
      </div>
    </div>

    <CategoryOverviews cities={props.cities} />
  </PageWrapper>
);

AllCitiesOverview.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    indicators: PropTypes.object.isRequired,
  })).isRequired,
};

export default AllCitiesOverview;
