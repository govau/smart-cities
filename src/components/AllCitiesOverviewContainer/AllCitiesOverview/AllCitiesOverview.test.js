import React from 'react';
import { shallow } from 'enzyme';
import AllCitiesOverview from './AllCitiesOverview';

jest.mock('../../CategoryOverviews/CategoryOverviews', () => 'CategoryOverviews');
jest.mock('../../Icon/Icon', () => 'Icon');

const defaultProps = {
  cities: [
    {
      id: 'perth',
      indicators: {},
    },
    {
      id: 'sydney',
      indicators: {},
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <AllCitiesOverview {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
