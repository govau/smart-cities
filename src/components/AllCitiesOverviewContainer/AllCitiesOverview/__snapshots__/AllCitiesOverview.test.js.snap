// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper categoryId=\\"overview\\" cityId=\\"all-cities\\">
  <div className=\\"heroWrapper\\">
    <div className=\\"heroContainer\\">
      <p className=\\"heroText\\">
        Tracking progress and performance across Australia’s largest cities
      </p>
      <img className=\\"smartCitiesLogo\\" src=\\"smart-cities-logo-white.svg\\" alt=\\"\\" />
    </div>
  </div>
  <div className=\\"infoWrapper\\">
    <div className=\\"infoContainer\\">
      <h1 id=\\"focus-point\\" className=\\"infoTitle\\">
        About
      </h1>
      <p className=\\"infoBodyText\\">
        Australian cities have long been an important economic and social asset and will be more important than ever in the 21st century. Our cities provide great opportunities for innovation and productivity in the global knowledge-based economy and are recognised as great places to live, work and study.
      </p>
      <p className=\\"infoBodyText\\">
        The Australian Government is committed to the continual improvement of our cities. 
        <Link href=\\"https://infrastructure.gov.au/cities/national-cities-performance-framework/\\" unstyled={false}>
          The National Cities Performance Framework
        </Link>
         will assist in this goal by providing data to help all levels of government, industry and the community to make the best policy and investment decisions for Australia’s future.
      </p>
    </div>
  </div>
  <div className=\\"subInfoWrapper\\">
    <div className=\\"subInfoContainer\\">
      <h1 className=\\"infoTitle\\">
        <span className=\\"infoTitleHeavy\\">
          Policy priorities
        </span>
         &amp; Context
      </h1>
      <p className=\\"subInfoBodyText\\">
        There are 6 policy priorities to help achieve the objectives of the 
        <Link href=\\"https://infrastructure.gov.au/cities/smart-cities/plan/index.aspx\\" unstyled={false}>
          Smart Cities Plan
        </Link>
        . The National Cities Performance Framework is designed to measure how well our cities are performing, using two types of indicators.
      </p>
      <dl className=\\"legend\\">
        <div className=\\"legendRow\\">
          <dt className=\\"term\\">
            <Icon className=\\"icon\\" color=\\"GREY_600\\" icon=\\"performanceIndicator\\" size={32} />
            Performance
          </dt>
          <dd className=\\"definition\\">
            Aspects that can be measured and tracked against policy priorities.
          </dd>
        </div>
        <div className=\\"legendRow\\">
          <dt className=\\"term\\">
            <Icon className=\\"icon\\" color=\\"GREY_600\\" icon=\\"contextualIndicator\\" size={32} />
            Context
          </dt>
          <dd className=\\"definition\\">
            Characteristics that help us understand what it’s like and why it functions the way it does.
          </dd>
        </div>
      </dl>
    </div>
  </div>
  <CategoryOverviews cities={{...}} />
</PageWrapper>"
`;
